files/image_config/corefile_uploader/core_uploader.py

```diff
@@ -221,18 +229,18 @@ def handle_file(path):
         tar.close()
         log_debug("Tar file for upload created: " + tarf_name)
 
-        Handler.upload_file(tarf_name, tarf_name)
+        Handler.upload_file(tarf_name, tarf_name, path)
 
         log_debug("File uploaded - " +  path)
         os.chdir(INIT_CWD)
 
     @staticmethod
-    def upload_file(fname, fpath):
+    def upload_file(fname, fpath, coref):
         daemonname = fname.split(".")[0]
         i = 0
         fail_msg = ""
         
-        while i <= MAX_RETRIES:
+        while True:
```

Comment:
I think you can do retry in the big loop. if it fails, retry in the big loop.

Reply:
I guess no. The bigger loop is either scan/wait-for-core. In either case, the next one would suffer the same fate. So I rather spew log & retry, until either I succeed or service is restarted, by someone alerted by these log messages.